scripts/actions/mobskills/impact_stream.lua

```diff
@@ -13,14 +13,22 @@ mobskillObject.onMobSkillCheck = function(target, mob, skill)
 end
 
 mobskillObject.onMobWeaponSkill = function(target, mob, skill)
-    local damage = math.floor(mob:getWeaponDmg() * 2.5)
-
-    damage = xi.mobskills.mobMagicalMove(mob, target, skill, damage, xi.element.LIGHT, 1, 0, 1)
+    local damage = xi.mobskills.mobMagicalMove(mob, target, skill, mob:getMainLvl() * 2, xi.element.NONE, 1, xi.mobskills.magicalTpBonus.NO_EFFECT)
     damage = xi.mobskills.mobFinalAdjustments(damage, mob, skill, target, xi.attackType.MAGICAL, xi.damageType.EARTH, xi.mobskills.shadowBehavior.WIPE_SHADOWS)
 
-    target:takeDamage(damage, mob, xi.attackType.MAGICAL, xi.damageType.EARTH)
+    target:takeDamage(damage, mob, xi.attackType.MAGICAL, xi.damageType.ELEMENTAL)
     xi.mobskills.mobStatusEffectMove(mob, target, xi.effect.STUN, 1, 0, 4)
-    xi.mobskills.mobStatusEffectMove(mob, target, xi.effect.DEFENSE_DOWN, 50, 0, 60)
+
+    -- Defense down power scales with TP
+    local power = 50
+    local tp = skill:getTP()
+    if tp == 3000 then
+        power = 60
+    elseif tp >= 2000 then
+        power = 55
+    end
```

Comment:
`local power = 50 + 5 * math.floor((skill:getTp() - 1000) / 1000)`

Reply:
Got that put in, good to know for the future. Thanks.